src/recordlinker/schemas/pii.py

```diff
@@ -163,6 +163,22 @@ class Address(pydantic.BaseModel):
     latitude: typing.Optional[float] = None
     longitude: typing.Optional[float] = None
 
+    @pydantic.field_validator("state", mode="before")
+    def normalize_state(cls, value):
+        """
+        Normalize the state field into 2-digit USPS code.
+        """
+        if value:
+            value = value.strip().title()
+
+            if len(value) == 2 and value.upper() in _STATE_CODE_TO_NAME:
+                return value.upper()
+
+            if value in _STATE_NAME_TO_CODE:
```

Comment:
I think this search also needs to be case-insensitive.  I'm thinking what if the incoming payload has `"state": "oregon"`.